pkg/geoipupdate/retry/retry.go

```diff
@@ -0,0 +1,31 @@
+package retry
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+)
+
+// MaybeRetryRequest is an internal implementation detail of this module. It
+// shouldn't be used by users of the geoipupdate Go library. You can use the
+// RetryFor field of geoipupdate.Config if you'd like to retry failed requests
+// when using the library directly.
+func MaybeRetryRequest(c *http.Client, retryFor time.Duration, req *http.Request) (*http.Response, error) {
```

Comment:
Reusing http.Requests is not generally possible. For example if there's a body. Perhaps we could error in that case if we want to do this?